service/src/test/java/io/camunda/service/query/filter/UserTaskFilterTest.java

```diff
@@ -434,4 +435,57 @@ public void shouldQueryByDueDateAndFollowUpDate() {
     assertThat(queryVariant).isInstanceOf(SearchBoolQuery.class);
     assertThat(((SearchBoolQuery) queryVariant).must()).hasSize(3);
   }
+
+  @Test
+  public void shouldQueryWithinPriorityRange() {
+    // given
+    final var priorityFilter = FilterBuilders.userTask().priority((f) -> f.gt(10L).lt(100L));
+    final var searchQuery =
+        SearchQueryBuilders.userTaskSearchQuery((b) -> b.filter(priorityFilter.build()));
+    // when
+    services.search(searchQuery);
+    // then
+    final var searchRequest = client.getSingleSearchRequest();
+    final var queryVariant = searchRequest.query().queryOption();
+
+    assertThat(queryVariant).isInstanceOf(SearchBoolQuery.class);
+    assertThat(queryVariant)
+        .isInstanceOfSatisfying(
+            SearchBoolQuery.class,
+            (t) ->
+                assertThat(t.must().get(1).queryOption())
+                    .isInstanceOfSatisfying(
+                        SearchRangeQuery.class,
+                        (term) -> {
+                          assertThat(term.field()).isEqualTo("priority");
+                          assertThat(term.gt()).isEqualTo(10L);
+                          assertThat(term.lt()).isEqualTo(100L);
+                        }));
+  }
+
+  @Test
+  public void shouldQueryWithPriority() {
+    // given
+    final var priorityFilter = FilterBuilders.userTask().priority((f) -> f.eq(10L));
+    final var searchQuery =
+        SearchQueryBuilders.userTaskSearchQuery((b) -> b.filter(priorityFilter.build()));
+    // when
+    services.search(searchQuery);
+    // then
+    final var searchRequest = client.getSingleSearchRequest();
+    final var queryVariant = searchRequest.query().queryOption();
+
+    assertThat(queryVariant).isInstanceOf(SearchBoolQuery.class);
```

Comment:
🔧  Is it possible add the test cases other operations for gt,... ? 

In order to check if the filter is applied for other cases

Or the scope is only equals from now?